### DISCUSSION TITLE
Running events from both parent and child state on the same event

### DISCUSSION THREAD

**Discussion by cristianpjensen** [statelyai/xstate#3857] [Q&A] [answered]
Hi. First time asking a question here. :-)

I want to always run some action on an event, so I have defined the event with its action in the root state. But, in a child state, I also want to run an action on that event. The default behaviour is for only the action in the child state to run. But, I need the action in the root state to _always_ run, even when the event is also defined in a child state. This behaviour is described [in the docs](https://xstate.js.org/docs/guides/hierarchical.html#events), but it does not say how to override this behaviour. Also in the [stately docs on xstate](https://stately.ai/docs/xstate/states/parent-and-child-states#on-in-parent-states) it only describes the default behaviour and not how to override it.

Does anyone know how to do this?

**Comment by davidkpiano** [comment, marked as answer]
Does the order of the actions matter?

Parallel states can help here, since multiple transitions can occur simultaneously:

```ts
const machine = createMachine({
  type: 'parallel',
  states: {
    watcher: {
      on: {
        EVENT: { actions: 'doSomething' }
      }
    },
    otherLogic: {
      // ...
      states: {
        someState: {
          on: {
            EVENT: {
              actions: 'doSomethingElse'
              // ...
            }
          }
        }
      }
    }
  }
});
```

Both `doSomething` and `doSomethingElse` will be executed on the `EVENT` transition in state `someState`.

It seems a little verbose, but it actually makes sense architecturally; you have two orthogonal "concerns" in your statechart. Sort of like the [sidecar pattern](https://learn.microsoft.com/en-us/azure/architecture/patterns/sidecar) in microservice architecture.

**Comment by cristianpjensen** [comment]
Thanks, that helped a lot.
